pango.c

```diff
@@ -93,15 +93,17 @@ PangoLayout *get_pango_layout(cairo_t *cairo, const char *font,
 
 void get_text_size(cairo_t *cairo, const char *font, int *width, int *height,
 		int *baseline, double scale, bool markup, const char *fmt, ...) {
-	char *buf = malloc(max_chars);
+	va_list args;
+
+	uint32_t size = vsnprintf(NULL, 0, fmt, args);
+	char *buf = malloc(size);
```

Comment:
Need to allocate one byte more, for the terminating NULL byte (see the manpage).

Reply:
I see, `vsnprintf`s return value excludes the NULL byte.

Reply:
Yeah, that's not the best API ever. :S